src/project.cc

```diff
@@ -181,17 +189,17 @@ Project::Entry GetCompilationEntryFromCompileCommandEntry(
 
   // Clang does not have good hueristics for determining source language, we
   // should explicitly specify it.
-  if (!AnyStartsWith(result.args, "-x")) {
-    if (IsCFile(entry.file))
-      result.args.push_back("-xc");
-    else
-      result.args.push_back("-xc++");
-  }
-  if (!AnyStartsWith(result.args, "-std=")) {
-    if (IsCFile(entry.file))
-      result.args.push_back("-std=c11");
-    else
-      result.args.push_back("-std=c++11");
+  if (auto sourceFileType = SourceFileType(entry.file)) {
+    if (AnyStartsWith(result.args, "-x")) {
+      result.args.push_back("-x" + *sourceFileType);
+    }
+    if (!AnyStartsWith(result.args, "-std=")) {
+      if (*sourceFileType == "c")
```

Comment:
This feels like it should be an enum, with a ToString() method for the scenario above. Might be too much boilerplate though.